Add support for MPI builds with the GNU compiler on Gadi

The build script currently does not support MPI builds with the GNU compiler on Gadi. Executing `./build.bash --compiler gnu --mpi` on Gadi produces the following error when compiling MPI source files:

```
f951: Fatal Error: Reading module ‘/apps/intel/compilers_and_libraries_2019.5.281/linux/mpi/intel64/include/mpi.mod’ at line 1 column 2: Unexpected EOF
compilation terminated.
gmake[2]: *** [CMakeFiles/cable-mpi.dir/build.make:75: CMakeFiles/cable-mpi.dir/src/offline/cable_mpicommon.F90.o] Error 1
```

This is due to the `intel-mpi` module being loaded when we should be using an MPI module that is compatible with the chosen compiler. See below:

https://github.com/CABLE-LSM/CABLE/blob/46830b3773f1932680af158cab27ae223fd8685a/build.bash#L92-L94

--- a/build.bash
+++ b/build.bash
@@ -75,10 +75,12 @@ if hostname -f | grep gadi.nci.org.au > /dev/null; then
         intel)
             module add intel-compiler/2019.5.281
             compiler_lib_install_dir=Intel
+            [[ -n $mpi ]] && module add intel-mpi/2019.5.281
             ;;
         gnu)
             module add gcc/13.2.0
             compiler_lib_install_dir=GNU
+            [[ -n $mpi ]] && module add openmpi/4.1.4
             ;;
         ?*)
             echo -e "\nError: compiler ${compiler} is not supported.\n"
@@ -88,10 +90,6 @@ if hostname -f | grep gadi.nci.org.au > /dev/null; then
     # This is required so that the netcdf-fortran library is discoverable by
     # pkg-config:
     prepend_path PKG_CONFIG_PATH "${NETCDF_BASE}/lib/${compiler_lib_install_dir}/pkgconfig"
-
-    if [[ -n $mpi ]]; then
-        module add intel-mpi/2019.5.281
-    fi
 
     if module is-loaded openmpi; then
         # This is required so that the openmpi MPI libraries are discoverable

